src/coreclr/jit/lclmorph.cpp

```diff
@@ -174,26 +139,21 @@ class LocalSequencer final : public GenTreeVisitor<LocalSequencer>
     // Arguments:
     //     node - The node
     //
-    void MoveNodeToEnd(GenTree* node)
+    void MoveNodeToEnd(GenTreeLclVarCommon* node)
     {
-        if (node->gtNext == nullptr)
+        if ((m_prevNode == node) || (node->gtNext == nullptr))
```

Comment:
Is `node->gtNext == nullptr` a meaningful check if `SequenceLocal` no longer unconditionally clears it?

Reply:
I had to recheck, but it turns out the answer is yes. The scenario is when we have half-consistent state during local morph with `m_stmtModified == true` and a new (bashed to) local address node inserted into the tree. Arguably, we should not be sequencing locals once `m_stmtModified` becomes `true`, but that is pre-existing behavior.